service/controller/app/v1/resource/chart/create.go

```diff
@@ -0,0 +1,28 @@
+package chart
+
+import (
+	"context"
+	"fmt"
+	"github.com/giantswarm/app-operator/service/controller/app/v1/key"
+	"github.com/giantswarm/microerror"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
+	customResource, err := key.ToCustomResource(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	name := key.AppName(customResource)
+	chart, err := r.g8sClient.ApplicationV1alpha1().Charts(r.watchNamespace).Get(name, v1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
```

Comment:
Can you check how we prefer to handle errors here: https://github.com/giantswarm/fmt/blob/master/go/errors.md#matching-errors.

Reply:
If you are talking about returning variables, I deliberately return it as nil since no instance I can find.

Reply:
I mean the conditional indentation. We try to keep it straight forward and test one matcher after another in case there are multiple ones. Let's hang if this is not clear.